event_sourcery_sqlalchemy/event_store.py

```diff
@@ -103,17 +104,24 @@ def _ensure_stream(self, stream_id: StreamId, versioning: Versioning) -> None:
         matching_streams_stmt = select(StreamModel).where(condition)
         matching_streams = self._session.execute(matching_streams_stmt).scalars().all()
         if not matching_streams:
-            ensure_stream_stmt = (
-                postgresql_insert(StreamModel)
-                .values(
-                    uuid=stream_id,
-                    name=stream_id.name,
-                    category=stream_id.category or "",
-                    version=initial_version,
+            values = {
+                "uuid": stream_id,
+                "name": stream_id.name,
+                "category": stream_id.category or "",
+                "version": initial_version,
+            }
+            if self._session.bind.dialect.name != "mysql":
```

Comment:
I would extract this part. I think we should avoid dialect-dependent elements. When we can't then we should limit escalation of it.

Reply:
true that, although... it's the only such place needed for now. But I get your point